components/script/dom/event.rs

```diff
@@ -102,6 +102,36 @@ impl Event {
         self.bubbles.set(bubbles);
         self.cancelable.set(cancelable);
     }
+    
+    // Determine if there are any listeners for a given target and type.
+    // See https://github.com/whatwg/dom/issues/453
+    pub fn has_listeners_for(&self, target: &EventTarget, type_: &Atom) -> bool {
```

Comment:
As a note, I intend to implement "has any listeners" more fully and with additional tests, perhaps together with adding a 'removed' flag to listeners, based on suggestions here whatwg/html#3646 (comment), and I'd rather do it in a follow-up. The current implementation is good enough to make the current tests pass(so it's equivalent to my previous hack).